### PR TITLE
[spark-rapids] Prevent automatic updates for Nvidia packages

### PR DESCRIPTION
**Pull request by SurajAralihalli**
This PR uses `apt-mark hold` to prevent auto upgrades for Nvidia packages.

signed-off-by: Suraj Aralihalli <suraj.ara16@gmail.com>

### CODE REVIEW

**Comment by cjac** [comment]
nice.  Tests are passing.

File: spark-rapids/spark-rapids.sh [hunk 1]
```diff
@@ -340,6 +340,16 @@ EOF
 readonly NVIDIA_BASE_DL_URL='https://developer.download.nvidia.com/compute'
 readonly NVIDIA_REPO_URL="${NVIDIA_BASE_DL_URL}/cuda/repos/${shortname}/x86_64"
 
+# Hold all NVIDIA-related packages from upgrading unintenionally or services like unattended-upgrades
+# Users should run apt-mark unhold before they wish to upgrade these packages
+function hold_nvidia_packages() {
+  apt-mark hold nvidia-*
+  apt-mark hold libnvidia-*
+  if apt list --installed 2>/dev/null | grep -q "xserver-xorg-video-nvidia"; then
```
Comment (cjac) [inline spark-rapids/spark-rapids.sh:348]: `dpkg -l` is lighter weight than `apt list --installed` ; I prefer not to use 'apt' and instead prefer 'apt-get' when it must be used in automation.  The ABI of apt is not stable.

Comment (SurajAralihalli) [reply spark-rapids/spark-rapids.sh]: Updated, thanks!

**Review by cjac — APPROVED** [review]
LGTM

**Pull request merged**